src/Prima/Pyxis/Modal.elm

```diff
@@ -0,0 +1,360 @@
+module Prima.Pyxis.Modal exposing
+    ( Config, config
+    , ModalSize, small, medium, large
+    , ModalHeader, defaultHeader, customHeader
+    , ModalFooter, emptyFooter, withButtonsFooter, customFooter
+    , view
+    )
+
+{-| Allows to create a Modal component using predefined Html syntax.
+
+
+# Modal Configuration
+
+@docs Config, config
+
+
+# Modal Size
+
+@docs ModalSize, small, medium, large
+
+
+# Modal Header
+
+@docs ModalHeader, defaultHeader, customHeader
+
+
+# Modal Footer
+
+@docs ModalFooter, emptyFooter, withButtonsFooter, customFooter
+
+
+# Rendering
+
+@docs view
+
+-}
+
+import Html exposing (..)
+import Html.Attributes exposing (class, classList)
+import Html.Events exposing (onClick)
+import Json.Decode
+import Prima.Pyxis.Button as Button
+
+
+{-| Represents the configuration of a Modal component.
+The Modal will trigger a "closed" Msg when the overlay is clicked.
+Is up to you to intercept and manage this Msg.
+-}
+type Config msg
+    = Config (Configuration msg)
+
+
+type alias Configuration msg =
+    { size : ModalSize
+    , header : ModalHeader msg
+    , content : List (Html msg)
+    , footer : ModalFooter msg
+    , closeTagger : msg
+    }
+
+
+{-| Creates a Modal.Config.
+
+    --
+    import Prima.Pyxis.Modal as Modal
+    import Html exposing (Html, p, text)
+
+    ...
+
+    type Msg =
+        ModalClosed
+
+    ...
+
+    myModalConfig : Modal.Config
+    myModalConfig =
+        let
+            modalHeader : Modal.ModalHeader Msg
+            modalHeader =
+                Modal.defaultHeader "My modal title"
+
+            modalFooter : Modal.ModalFooter Msg
+            modalFooter =
+                Modal.emptyFooter
+
+            modalContent : List (Html Msg)
+            modalContent =
+                p [] [ text "Lorem ipsum dolor sit amet."]
+        in
+        Modal.config Modal.medium modalHeader  modalContent modalFooter ModalClosed
+
+-}
+config : ModalSize -> ModalHeader msg -> List (Html msg) -> ModalFooter msg -> msg -> Config msg
+config size header content footer closeTagger =
+    Config <| Configuration size header content footer closeTagger
+
+
+{-| Represents the Modal size.
+-}
+type ModalSize
+    = SmallModalSize
+    | MediumModalSize
+    | LargeModalSize
+
+
+{-| Creates a Small ModalSize
+-}
+small : ModalSize
+small =
+    SmallModalSize
+
+
+{-| Creates a Medium ModalSize
+-}
+medium : ModalSize
+medium =
+    MediumModalSize
+
+
+{-| Creates a Large ModalSize
+-}
+large : ModalSize
+large =
+    LargeModalSize
+
+
+isSmallModalSize : ModalSize -> Bool
+isSmallModalSize =
+    (==) SmallModalSize
+
+
+isMediumModalSize : ModalSize -> Bool
+isMediumModalSize =
+    (==) MediumModalSize
+
+
+isLargeModalSize : ModalSize -> Bool
+isLargeModalSize =
+    (==) LargeModalSize
+
+
+{-| Represents the Header configuration of a Modal.
+-}
+type ModalHeader msg
+    = ModalHeaderDefault String
+    | ModalHeaderCustom (List (Html msg))
```

Comment:
anche qui potrebbe essere `Html msg`?

Reply:
Idem come sopra. In ogni caso l'importante è che abbiano la stessa firma.